pubtools/pulplib/_impl/model/repository/yum.py

```diff
@@ -1,9 +1,86 @@
-from .base import Repository, repo_type
+from .base import Repository, SyncOptions, repo_type
 from ..frozenlist import FrozenList
 from ..attr import pulp_attrib
 from ... import compat_attr as attr
 
 
+@attr.s(kw_only=True, frozen=True)
+class YumSyncOptions(SyncOptions):
+    """Options controlling a container repository
+    :meth:`~pubtools.pulplib.YumRepository.sync`.
+    """
+
+    query_auth_token = pulp_attrib(default=None, type=str)
+    """An authorization token that will be added to every request made to the feed URL's server
+    """
+
+    validate = pulp_attrib(default=None, type=bool)
+    """If True, checksum of each file will be verified against the metadata's expectation
+    """
+
+    max_downloads = pulp_attrib(default=None, type=int)
+    """Number of threads used when synchronizing the repository.
+    """
+
+    remove_missing = pulp_attrib(default=None, type=bool)
+    """If true, as the repository is synchronized, old rpms will be removed.
+    """
+
+    retain_old_count = pulp_attrib(default=None, type=int)
+    """Count indicating how many old rpm versions to retain.
+    """
+
+    skip = pulp_attrib(default=None, type=list)
+    """List of content types to be skipped during the repository synchronization
+    """
+
+    checksum_type = pulp_attrib(default=None, type=str)
+    """checksum type to use for metadata generation.
+
+    Defaults to source checksum type of sha256
+    """
+
+    num_retries = pulp_attrib(default=None, type=int)
+    """Number of times to retry before declaring an error during repository synchronization
+
+    Default is to 2.
+    """
+
+    copy_children = pulp_attrib(default=None, type=bool)
+    """when False, will not attempt to locate and copy child packages of errata, groups, or categories
+    """
+
+    download_policy = pulp_attrib(default=None, type=str)
+    """Set the download policy for a repository.
+
+    Supported options are immediate,on_demand,background
+    """
+
+    force_full = pulp_attrib(default=None, type=bool)
+    """Boolean flag. If true, full re-sync is triggered.
+    """
+
+    require_signature = pulp_attrib(default=None, type=bool)
+    """Requires that imported packages like RPM/DRPM/SRPM should be signed
+    """
+
+    allowed_keys = pulp_attrib(default=None, type=str)
+    """Comma-separated list of allowed signature key IDs that imported packages can be signed with
+    """
+
+    recursive = pulp_attrib(default=None, type=bool)
```

Comment:
I don't think this one makes sense either.  I am pretty sure recursive applies during copy only, not sync.  In general can we please only include options here if we understand the use-case for them?